batch/batch/driver/pool.py

```diff
@@ -428,16 +437,25 @@ def get_instance(user, cores_mcpu):
                 instance = self.pool.healthy_instances_by_free_cores[i]
                 assert cores_mcpu <= instance.free_cores_mcpu
                 if user != 'ci' or (user == 'ci' and instance.zone.startswith('us-central1')):
+                    if user == 'ci':
+                        self.ci_no_viable_instances.value = 0
                     return instance
                 i += 1
+
             histogram = collections.defaultdict(int)
             for instance in self.pool.healthy_instances_by_free_cores:
                 histogram[instance.free_cores_mcpu] += 1
             log.info(f'schedule {self.pool}: no viable instances for {cores_mcpu}: {histogram}')
+
+            if user == 'ci':
+                self.ci_no_viable_instances.value += 1
+
             return None
 
         should_wait = True
-        for user, resources in user_resources.items():
+        users = sorted(user_resources.keys(), key=lambda user: user != 'ci')
+        for user in users:
+            resources = user_resources[user]
```

Comment:
this all seems like a mixing of concerns, the scheduler shouldn't have to inform the scaler that CI needs more instances.